scipy/sparse/linalg/tests/test_interface.py

```diff
@@ -284,6 +284,20 @@ def always_four_ones(x):
         assert_(hasattr(op, "shape"))
         assert_(hasattr(op, "_matvec"))
 
+def matvec(x):
+    """ Needed for test_pickle as local functions are not pickleable """
+    return np.zeros(3)
+
+def test_pickle():
+    import pickle
+
+    N = M = 3
+    A = interface.LinearOperator((M, N), matvec)
+    s = pickle.dumps(A)
+    B = pickle.loads(s)
+
+    for k in A.__dict__:
+        assert_equal(getattr(A, k), getattr(B, k))
```

Comment:
I just came to a similar conclusion myself. Given that it really isn't optimal to have two places to change code when one of that arguments to `__init__` changes how does that leave this PR. Only pass through the required arguments (shape and matvec) in `__getnewargs__`?